diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2017-09-29 10:42:33.000000000 +0200
+++ b/CMakeLists.txt	2018-03-11 01:16:51.784496593 +0100
@@ -832,10 +832,15 @@
   set(libdir      ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
   set(includedir  ${CMAKE_INSTALL_PREFIX}/include)
   set(LIBS        "-lz -lm")
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(PNGLIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+  endif()
+  file(REMOVE ${CMAKE_CURRENT_BINARY_DIR}/${PNGLIB_NAME}.pc)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in
     ${CMAKE_CURRENT_BINARY_DIR}/${PNGLIB_NAME}.pc @ONLY)
   CREATE_SYMLINK(libpng.pc FILE ${PNGLIB_NAME}.pc)

+  file(REMOVE ${CMAKE_CURRENT_BINARY_DIR}/${PNGLIB_NAME}-config)
   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libpng-config.in
     ${CMAKE_CURRENT_BINARY_DIR}/${PNGLIB_NAME}-config @ONLY)
   CREATE_SYMLINK(libpng-config FILE ${PNGLIB_NAME}-config)
diff -Naur a/libpng-config.in b/libpng-config.in
--- a/libpng-config.in	2017-09-29 10:42:33.000000000 +0200
+++ b/libpng-config.in	2018-03-11 01:17:16.264728990 +0100
@@ -16,8 +16,8 @@
 exec_prefix="@exec_prefix@"
 libdir="@libdir@"
 includedir="@includedir@/libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@"
-libs="-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@"
-all_libs="-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@ @LIBS@"
+libs="-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@@LIBPNG_POSTFIX@"
+all_libs="-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@@LIBPNG_POSTFIX@ @LIBS@"
 I_opts="-I${includedir}"
 L_opts="-L${libdir}"
 R_opts=""
diff -Naur a/libpng.pc.in b/libpng.pc.in
--- a/libpng.pc.in	2017-09-29 10:42:33.000000000 +0200
+++ b/libpng.pc.in	2018-03-11 01:17:08.736657534 +0100
@@ -7,6 +7,6 @@
 Description: Loads and saves PNG files
 Version: @PNGLIB_VERSION@
 Requires: zlib
-Libs: -L${libdir} -lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@
+Libs: -L${libdir} -lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@@PNGLIB_POSTFIX@
 Libs.private: @LIBS@
 Cflags: -I${includedir}
